SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -102,13 +126,15 @@ - (void)sdl_accessoryDisconnected:(NSNotification *)notification {
     EAAccessory *accessory = [notification.userInfo objectForKey:EAAccessoryKey];
     if (accessory.connectionID == self.session.accessory.connectionID) {
         self.sessionSetupInProgress = NO;
+        [self sdl_backgroundTaskEnd];
         [self disconnect];
         [self.delegate onTransportDisconnected];
     }
 }
 
 - (void)sdl_applicationWillEnterForeground:(NSNotification *)notification {
     [SDLDebugTool logInfo:@"App Foregrounded Event" withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+    [self sdl_backgroundTaskEnd];
```

Comment:
http://stackoverflow.com/questions/29919573/beginbackgroundtaskwithexpirationhandler-calling-endbackgroundtask-but-not-endin

In the above article, it appears that not explicitly ending the background task when the app foregrounds can lead to the app being terminated.